src/ml_board/frontend/newDashboard/src/webworkers/DedicatedWorkerClass.tsx

```diff
@@ -0,0 +1,42 @@
+type DedicatedWorkerClassInterface = {
+    worker: any // I don't know what type should come here. Ask someone regarding this...
+    userCallbacks: {
+        onMessageCtxNFunc: Function | null
+    }
+}
+
+class DedicatedWorkerClass implements DedicatedWorkerClassInterface{
+    
+    worker: any // I don't know what type should come here. Ask someone regarding this...
+    userCallbacks: {
+        onMessageCtxNFunc: Function | null
+    }
+
+    constructor(onMessageCtxNFunc = null) {
+        if(window.Worker) {
+            this.worker = new Worker(new URL('./worker.tsx', import.meta.url));
+            this.worker.onmessage = (e: any) => {
+                this.onMessage(e);
+            }
+            this.userCallbacks = {
+                onMessageCtxNFunc
+            }
+        }
+        else {
+            throw new Error('WebWorker not supported by browser. Please use an updated browser.');
+        }
+    }
+
+    postMessage(data = {}, transferData = []) {
+        this.worker.postMessage(data, transferData)
+    }
+
+    onMessage(e:any) {
+        if (this.userCallbacks !== null) {
+            this.userCallbacks.onMessageCtxNFunc && 
```

Comment:
Don't need "&&" now as onMessageCtxNFunc is made to be required always (otherwise data sent back from worker to main tread will be lost: mentioned in one of the comments also)